bin/install-qa-check.d/60pkgconfig

```diff
@@ -102,23 +102,35 @@ pkgconfig_check() {
 	if [[ -n ${QA_PKGCONFIG_VERSION} ]]; then
 		local all_bad=yes
 		# Record the last bad file matched
-		local bad_file
+		local -A bad_files
 		for f in "${files[@]}" ; do
-			if [[ ${QA_PKGCONFIG_VERSION} == $(pkg-config --modversion "${f}") ]] ; then
+			local file_version=$(pkg-config --modversion "${f}")
+			if [[ ${QA_PKGCONFIG_VERSION} == ${file_version} ]] ; then
 				all_bad=no
 				break
 			fi
 
-			bad_file="${f//${D}}"
+			# Record a special value if the .pc file has no version set at all.
+			if [[ -z ${file_version} ]]; then
+				file_version="<no-set>"
+			fi
+			bad_files["${f//${D}}"]="${file_version}"
 		done
 
 		# Skip result reporting if *_p* because for both _pN and _preN, we
 		# don't generally expect the versions to be exactly accurate, and
 		# we want to avoid false positives.
 		if [[ ${all_bad} == "yes" && ${PV} != *_p* ]] && ! has live ${PROPERTIES} ; then
 			eqawarn "QA Notice: pkg-config files with mismatched Version found!"
-			eqawarn "At least ${bad_file}'s Version field does not match ${PV}"
+			eqawarn "The following file's Version field does not match ${PV}"
+			local bad_file
+			for bad_file in "${!bad_files[@]}"; do
+				local bad_file_version="${bad_files[${bad_file}]}"
```

Comment:
See if there's a way to use eqatag? (https://www.gentoo.org/glep/glep-0065.html#eqatag)

(I'm guessing no.)

Reply:
Why are you guessing 'no'. My naive assumption would be that, given that `eqawarn` is available, `eqatag` would also be available here. and we can simply

```
eqatag pkgconfig.unexpected-version ${bad_files[@]}
local bad_file
for bad_file in "${!bad_files[@]}"; do
    …
```

Reply:
I think I'd assumed the output wouldn't fit and I think I assumed wrong :)